FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+                {
+                    // extract patient ids from group
+                    _logger.LogInformation($"Start extracting patients from group '{_filterConfiguration.GroupId}'.");
+
+                    // For now, the queryParameters is always null.
+                    // This parameter will be used when we enable filter groups in the future.
+                    var patientsHash = (await _groupMemberExtractor.GetGroupPatientsAsync(
+                        _filterConfiguration.GroupId,
+                        null,
+                        _inputData.DataEndTime,
+                        cancellationToken)).Select(TableKeyProvider.CompartmentRowKey).ToHashSet();
+
+                    var processedPatientVersions = await GetPatientVersions(patientsHash, cancellationToken);
+
+                    // set the version id for processed patient
+                    // the processed patients is empty at the beginning , and will be updated when completing a successful job.
+                    toBeProcessedPatients = patientsHash.Select(patientHash =>
+                        new PatientWrapper(
+                            patientHash,
+                            processedPatientVersions.ContainsKey(patientHash) ? processedPatientVersions[patientHash] : 0)).ToList();
+
+                    _logger.LogInformation(
+                        "Extract {patientCount} patients from group '{groupId}', including {newPatientCount} new patients.",
+                        patientsHash.Count,
+                        _filterConfiguration.GroupId,
+                        toBeProcessedPatients.Where(p => p.VersionId == 0).ToList().Count);
+                }
+
+                while (true)
+                {
+                    while (_result.RunningJobIds.Count > _schedulerConfiguration.MaxConcurrencyCount)
+                    {
+                        await WaitRunningJobComplete(progress, cancellationToken);
+                    }
+
+                    FhirToDataLakeProcessingJobInputData input = null;
+
+                    switch (_filterConfiguration.FilterScope)
+                    {
+                        case FilterScope.System:
+                            var interval = TimeSpan.FromSeconds(IncrementalOrchestrationIntervalInSeconds);
+
+                            if (_inputData.DataStartTime == null || ((DateTimeOffset)_inputData.DataStartTime).AddMinutes(60) < _inputData.DataEndTime)
+                            {
+                                interval = TimeSpan.FromSeconds(InitialOrchestrationIntervalInSeconds);
+                            }
+
+                            if (_result.NextJobTimestamp == null || _result.NextJobTimestamp < _inputData.DataEndTime)
+                            {
+                                var nextResourceTimestamp =
+                                    await GetNextTimestamp(_result.NextJobTimestamp ?? _inputData.DataStartTime, _inputData.DataEndTime, cancellationToken);
```

Comment:
I would suggest to return async IEnumerable here.

Reply:
async IEnumerable used.